bids_conversion/convert_somato_data.py

```diff
@@ -0,0 +1,255 @@
+"""Convert the MNE-somato-data to BIDS format.
+
+Requirements
+------------
+The Python dependencies to run this script are:
+
+- numpy>=1.16
+- scipy>=1.2
+- mne<0.19
+- mne-bids>=0.3
+- nibabel>=2.4.1
+
+FreeSurfer
+----------
+Next to the Python requirements, you will need FreeSurfer installed on your
+system. Once FreeSurfer is installed and you want to run this script, make sure
+to initialize FreeSurfer first:
+
+$ export FREESURFER_HOME=/path/to/FreeSurfer
+$ source $FREESURFER_HOME/SetUpFreeSurfer.sh
+
+Helpful links
+-------------
+- https://mne-tools.github.io/stable/index.html
+- https://mne-tools.github.io/mne-bids/
+- https://mne-tools.github.io/stable/manual/datasets_index.html?#somatosensory
+- https://surfer.nmr.mgh.harvard.edu/fswiki/FreeSurferWiki
+- https://mne-tools.github.io/stable/auto_tutorials/source-modeling/plot_background_freesurfer.html  # noqa: E501
+
+"""
+# Authors: Stefan Appelhoff <stefan.appelhoff@mailbox.org>
+#          Mainak Jas <mainakjas@gmail.com>
+#
+# License: BSD (3-clause)
+import os
+import os.path as op
+import shutil as sh
+import json
+
+import numpy as np
+
+import mne
+from mne.utils import check_version, run_subprocess
+import mne.datasets.somato as somato
+
+from mne_bids import write_raw_bids, make_bids_basename, write_anat
+from mne_bids.utils import print_dir_tree
+
+# Create READMEs for the BIDS dataset
+# Prepare the text for the central README
+main_readme = """MNE-somato-data-bids
+====================
+
+This dataset contains the MNE-somato-data in BIDS format.
+
+The conversion can be reproduced through the Python script stored in the
+`/code` directory of this dataset. See the README in that directory.
+
+The `/derivatives` directory contains the outputs of running the FreeSurfer
+pipeline `recon-all` on the MRI data with no additional commandline options
+(only defaults were used):
+
+$ recon-all -i sub-01_T1w.nii.gz -s sub-01 -all
+
+After the `recon-all` call, there were further FreeSurfer calls from the MNE
+API:
+
+$ mne make_scalp_surfaces -s sub-01 --force
+$ mne watershed_bem -s sub-01
+
+The derivatives also contain the forward model `*-fwd.fif`, which was produced
+using the source space definition, a `*-trans.fif` file, and the boundary
+element model (=conductor model) that lives in
+`freesurfer/subjects/sub-01/bem/*-bem-sol.fif`.
+
+The `*-trans.fif` file is not saved, but can be recovered from the anatomical
+landmarks in the `sub-01/anat/T1w.json` file and MNE-BIDS' function
+`get_head_mri_transform`.
+
+See: https://github.com/mne-tools/mne-bids for more information.
+
+Notes on FreeSurfer
+===================
+the FreeSurfer pipeline `recon-all` was run new for the sake of converting the
+somato data to BIDS format. This needed to be done to change the "somato"
+subject name to the BIDS subject label "01". Note, that this is NOT "sub-01",
+because in BIDS, the "sub-" is just a prefix, whereas the "01" is the subject
```

Comment:
I'm still confused by this, but I'm fine with merging this and see how this flies

Reply:
okay, let's merge!